jetty-websocket/websocket-core-common/src/main/java/org/eclipse/jetty/websocket/core/internal/messages/ByteBufferMessageSink.java

```diff
@@ -97,6 +96,7 @@ public void accept(Frame frame, Callback callback)
                 }
             }
 
+            callback.succeeded();
             session.demand(1);
```

Comment:
Should you demand if `frame.isFin()`?

Reply:
Yes we still need to demand, there is no demand in the Jetty WebSocket API.